data_steward/analytics/cdr_ops/README.md

```diff
@@ -4,4 +4,130 @@ Notebooks for operational evaluation of data.
+                                  This parameter has an unknown type.
+  my_string_parameter: str (default "", required=True)
+                                  This is a string parameter.
+  my_int_parameter: int (default 0, required=False)
+                                  This is an integer parameter.
+```
+
+### Running Notebook
+
+When executing a notebook, each parameter is passed with a separate <em>-p</em> or <em>--parameters</em> flag.
+
+```
+$ python report_runner.py "Example Notebook.py"  
+        --output_path executed_notebook.html  
+         -p my_unknown_parameter "hello"  
+         -p my_string_parameter "world"  
+         -p my_int_parameter 6
+Executing: 100%|█████████████████████████████████████████████████████| 3/3 [00:01<00:00,  2.11cell/s]
+```
+And the outputted html:
+![Executed HTML](./images/html_image.png)
+
+Notice that an extra cell containing the passed parameter values was injected into the notebook.
+
+### Debugging
+
+There are a few sets of errors one may encounter when attempting to execute a notebook.
+
+#### Missing/Unknown Parameters
+
+These errors occur when required parameters or values are not passed in or an unknown parameter is provided. A error message is logged and the notebook is not executed:
+
+```
+$ python report_runner.py "Example Notebook.py" 
+    --output_path executed_notebook.html 
+     -p my_unknown_parameter "hello"  
+     -p my_int_parameter 6
+Missing the parameter my_string_parameter for notebook Example Notebook
+Parameters inferred for notebook Example Notebook:
+  my_unknown_parameter: Unknown type (default "", required=True)
+                                  This parameter has an unknown type.
+  my_string_parameter: str (default "", required=True)
+                                  This is a string parameter.
+  my_int_parameter: int (default 0, required=False)
+                                  This is an integer parameter.
+```
+
+#### Failed Notebook Logic
+
+If an error occurs during the execution of the notebook cells, an error message will be logged <em>and</em> the error will be labeled in the resulting html file.
+
+```
+$ python report_runner.py "Example Notebook.py" --output_path executed_notebook.html -p my_unknown_parameter "hello" -p my_string_parameter "world" -p my_int_parameter 6
+Executing: 100%|█████████████████████████████████████████████████████| 3/3 [00:01<00:00,  2.08cell/s]
+
+---------------------------------------------------------------------------
+Exception encountered at "In [3]":
+---------------------------------------------------------------------------
+TypeError                                 Traceback (most recent call last)
+<ipython-input-3-6cf2cee3f746> in <module>()
+      1 print(f"My unknown type parameter is {my_unknown_parameter}")
+----> 2 print(f"My string parameter is {my_string_parameter / 0}")
+      3 print(f"My int parameter is {my_int_parameter}")
+
+TypeError: unsupported operand type(s) for /: 'str' and 'int'
```

Comment:
Verified that errors are presented when encountered. I needed to set the google credentials to a data_analytics key file.